canvasapi/requester.py

```diff
@@ -133,15 +133,18 @@ def _post_request(self, url, headers, data=None):
         :param data: dict
         """
 
-        # Grab file from data.
+        # Grab file/files from data.
```

Comment:
On my phone ATM, I'll try to look more into this later if necessary. If I recall correctly, sending files as an octet stream was the easiest way to support multiple file types (e.g. csv, zip, xml) without specifying them explicitly. You can see an example of this in the requests library documentation.

I don't believe sending it as a singular `file` will work without some additional header changes, and it would also require us to explicitly state the file type in the headers.

http://docs.python-requests.org/en/v1.2.3/user/quickstart/
https://canvas.instructure.com/doc/api/sis_imports.html#SisImport

Reply:
FWIW, I started working on about the same thing not realizing the feature request existed (and I see you're a lot further along than me). I modified request.py as:

```python
    ¦   # Grab file from data.
    ¦   file = None
    ¦   for field, value in data:
    ¦   ¦   if field == 'file':
    ¦   ¦   ¦   if isinstance(value, dict):
    ¦   ¦   ¦   ¦   file = value
    ¦   ¦   ¦   else:
    ¦   ¦   ¦   ¦   file = {'file': value}
    ¦   ¦   ¦   break
```

That allows specifying the full file-tuple using the same `file` parameter.